server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -2347,6 +2363,13 @@ static SortedMap<String, IndexAbstraction> buildIndicesLookup(
                         indexToDataStreamLookup.put(i.getName(), dsAbstraction);
                     }
                 }
+
+                for (SearchEngine searchEngine : searchEngineMetadata.searchEngines().values()) {
+                    assert searchEngine.getIndices().isEmpty() == false;
```

Comment:
Is there any rule of thumb for when to use assertions? For cluster related code?

Reply:
My reading of the existing code is that it is used in cases where you expect a code user to make the check before calling it (here we expect a search engine with a non-empty indices list).